sns/src/main/scala/akka/stream/alpakka/sns/SnsPublishSinkStage.scala

```diff
@@ -0,0 +1,77 @@
+package akka.stream.alpakka.sns
+
+import akka.Done
+import akka.stream.stage._
+import akka.stream.{Attributes, Inlet, SinkShape}
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sns.AmazonSNSAsync
+import com.amazonaws.services.sns.model.{PublishRequest, PublishResult}
+
+import scala.concurrent.{Future, Promise}
+
+final class SnsPublishSinkStage(topicArn: String, snsClient: AmazonSNSAsync)
```

Comment:
Yes the AmazonSNSAsync is ThreadSafe regarding to the AWS [docs.](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/sns/AmazonSNSAsyncClient.html) That's why I think it is fine to share a single AmazonSNSAsync among multiple materialization stages.

Also regarding your last comment about not making the AmazonSNSAsync implicit, I try to be consistent with other connectors, like the SQS one. There we also pass the AmazonSQSClient implicitly. I think it will be easier for people who use the connectors if the initialization of the GraphStages work the same way.

Good point to change the Sink to a Flow. I thought about keeping the Sink and provide an additional Flow GraphStage which will return the messageId of the published message, so clients can do further processing with the returned messageId.